make_atf.sh

```diff
@@ -48,10 +56,13 @@ while read Torig; do
 	atf_test_case $Tunder
 	${Tunder}_head() {
 	    atf_set timeout $timeout
+	    atf_set require.config allow_sysctl_side_effects
 	}
 	${Tunder}_body() {
+	    atf_check save_sysctls ${tcptestsuite_path}/${Torig}.pkt
 	    atf_check -o ignore ${packetdrill_path} \
 	        ${tcptestsuite_path}/${Torig}.pkt
+	    atf_check -o ignore sysctl -f saved_sysctls
```

Comment:
This will fail to restore the old sysctl values if the preceding atf_check command failed.  You should create a separate `${Tunder}_cleanup()` for that purpose.  ATF cleanup routines run regardless of whether the test passed or failed.

Reply:
Correct me if I'm wrong, but if I move this into a `_cleanup()` function, failure to restore the sysctls will be ignored.  I want the test to fail in that case, because a silent failure to restore them could lead to failures in later tests (even outside tcp-testsuite) that would be _very_ hard to diagnose because people might not think to look at sysctl values that the later test didn't modify.

Reply:
After a little more thought, I could restore the sysctls in _both_ places.  That would serve both purposes.

Reply:
Failing to restore sysctls after a test failure could _also_ lead to failures in later tests (even outside tcp-testsuite).  And a failure of a TCP test suite test is more likely than a failure to set a sysctl that has already been set once.

It's worth testing, but I think that ATF does not automatically ignore errors in the cleanup step.  IIRC an error during the cleanup step will make a test "broken", which is distinct from "failed".

Reply:
Man, I wish the ATF shell API had `atf_check` _and_ `atf_require`, like the C API.